### PR TITLE
Speedup random numbers

### PR DESCRIPTION
**Pull request by Anatoliy-Litvinenko**
Created  thread local random generated engines for each run iterations, making random generation thread independent. It make measurable performance improvement.  Also it should make result of computation more perfectly repeatable and more independent of number of parallel threads used.   (Immediate outcome: best run have same number with different attempts).  
Made easier to try other random engines. Current one is the fastest from standard library available.

### CODE REVIEW

File: input.cpp [hunk 1]
```diff
@@ -752,7 +752,6 @@ namespace marxan {
         tpf1 = 0;
         tpf2 = 0;
         repeats = 0;
-        asymmetricconnectivity = 0;
```
Comment (adelebai) [inline input.cpp]: whats' the reason for removing this?

**Review by adelebai — APPROVED** [review]

File: marxan.cpp [hunk 2]
```diff
@@ -442,8 +447,8 @@ namespace marxan {
 
         delta = 1e-14;  // This would more elegantly be done as a constant
 
-        // initt global rng engine
-        rngEngine = mt19937(iseed);
+        // init rng engine
```
Comment (adelebai) [inline marxan.cpp:450]: I think originally the RANDSEED if set to -1 means we set a seed based on machine time (or something similar) to intentionally vary the results. It only applies when the value is > 0. I can't recall if this is still the case.

**Review by andrist — APPROVED** [review]
Nice, looks good!

Just out of curiosity: what's the speedup you saw?

Would it be possible to add a test for the ability to reproduce the same result when fixing the global seed?

**Comment by Anatoliy-Litvinenko** [comment]
> Nice, looks good!
> 
> Just out of curiosity: what's the speedup you saw?
> 
> Would it be possible to add a test for the ability to reproduce the same result when fixing the global seed?

Speed up 10 sec out of 100 sec. (Where rand numbers take 40 sec). Not much, but really proud of making on the way the result to be deterministic and not dependent of number of processor cores and time of the day it runs. That is very important for catching and fixing bugs.

File: marxan.cpp (hunk 2 above)
Comment (Anatoliy-Litvinenko) [reply marxan.cpp:450]: rng_engine rngEngine(iseed); 
 below duplicate this functionality, where iseed  is read firm inputs.

File: input.cpp (hunk 1 above)
Comment (Anatoliy-Litvinenko) [reply input.cpp]: Undone it as not related to this pull request.

**Pull request merged**